JotunnLib/Extensions/GameObjectExtension.cs

```diff
@@ -104,6 +106,174 @@ public static GameObject OrNull(this GameObject @this)
+        /// <param name="gameObject"></param>
+        /// <param name="components"></param>
+        /// <returns></returns>
+        public static bool HasAnyComponentInChildren(
+            this GameObject gameObject,
+            bool includeInactive = false,
+            params Type[] components
+        )
+        {
+            foreach (var compo in components)
+            {
+                if (gameObject.GetComponentInChildren(compo, includeInactive) != null)
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        /// <summary>
+        ///     Check if GameObject or any of it's children
+        ///     have the specific component.
+        /// </summary>
+        /// <param name="gameObject"></param>
+        /// <param name="includeInactive">Whether to include inactive child objects in the search or not.</param>
+        /// <returns></returns>
+        public static bool HasComponentInChildren<T>(this GameObject gameObject, bool includeInactive = false) where T : Component
+        {
+            return gameObject.GetComponentInChildren<T>(includeInactive) != null;
+        }
+
+        /// <summary>
+        ///     Extension method to get the first component in the GameObject
+        ///     or it's children that has the specified name.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="gameObject"></param>
+        /// <param name="name"></param>
+        /// <param name="includeInactive">Whether to include inactive child objects in the search or not.</param>
+        /// <returns></returns>
+        public static T GetComponentInChildrenByName<T>(
+            this GameObject gameObject,
+            string name,
+            bool includeInactive = false
+        ) where T : Component
+        {
+            foreach (var compo in gameObject.GetComponentsInChildren<T>(includeInactive))
+            {
+                if (compo.name == name)
+                {
+                    return compo;
+                }
+            }
+            Logger.LogWarning($"No {nameof(T)} with name {name} found for GameObject: {gameObject.name}");
+            return null;
+        }
+
+        /// <summary>
+        ///     Extension method to find nested children by name using either
+        ///     a breadth-first or depth-first search. Default is breadth-first.
+        /// </summary>
+        /// <param name="gameObject"></param>
+        /// <param name="childName">Name of the child object to search for.</param>
+        /// <param name="breadthFirst"> Whether to preform a breadth first or depth first search. Default is breadth first.</param>
+        /// <returns></returns>
+        public static Transform FindDeepChild(this GameObject gameObject, string childName, bool breadthFirst = true)
+        {
+            return gameObject?.transform.FindDeepChild(childName, breadthFirst);
+        }
```

Comment:
Valheim already has the method `Utils.FindChild`, which does the same. Although Wrapping this into an extension method could still be done, it should call the vanilla method.

Also, the null conditional (.?) on the gameobect doesn't work as expedted because of the Unity lifecyle [(it can return true despite the object being destroyed)](https://stackoverflow.com/questions/62678228/why-does-c-sharp-null-conditional-operator-not-work-with-unity-serializable-vari) and should be removed anyway. It's not the purose if this method to test if the gameObject is null.

Reply:
Thanks for letting me know, will update accordingly and also very much appreciate the link to help learn more about Unity (will make sure to account for that in my own code now).